packages/alfa-rules/test/sia-er62/rule.spec.tsx

```diff
@@ -1070,229 +1225,255 @@ test(`evaluate() passes an <a> element that is distinguishable from the <p> pare
     [
       h.sheet([
         h.rule.style("p", {
-          backgroundImage: "linear-gradient(to right, #F9F9F1 50%, blue 50%)",
-          color: "rgba(255, 255, 255, 0.1)",
+          color: "#0000EE",
         }),
+      ]),
+    ]
+  );
 
-        h.rule.style("a, a:hover, a:focus", {
-          backgroundImage: "linear-gradient(to right, #F9F9F1 50%, blue 50%)",
+  t.deepEqual(await evaluate(ER62, { document }), [inapplicable(ER62)]);
+});
+
+test(`evaluate() is inapplicable to an <a> element with a <p> parent element
+    when both have the same foreground color and <a> element's sibling elements also have the same foreground color`, async (t) => {
+  const target = <a href="#">Link</a>;
+
+  const document = h.document(
+    [
+      <div>
+        <p>Hello</p>
+        <p>
+          <span>World</span>
+          Hello
+          <p>nested p</p>
+          {target}
+        </p>
+      </div>,
+    ],
+    [
+      h.sheet([
+        h.rule.style("a", {
           textDecoration: "none",
           outline: "none",
-          cursor: "auto",
+          borderWidth: "0",
+        }),
+        h.rule.style("p", {
+          color: "#0000EE",
+        }),
+        h.rule.style("span", {
+          color: "#0000EE",
         }),
       ]),
     ]
   );
 
-  const contrastPairings = [
-    Contrast.Pairing.of(
-      RGB.of(
-        Percentage.of(0.1),
-        Percentage.of(0.1),
-        Percentage.of(1),
-        Percentage.of(1)
-      ),
-      defaultLinkColor,
-      1.18
-    ),
-    Contrast.Pairing.of(
-      RGB.of(
-        Percentage.of(0.9788235),
-        Percentage.of(0.9788235),
-        Percentage.of(0.9505882),
-        Percentage.of(1)
-      ),
-      defaultLinkColor,
-      8.93
-    ),
-  ];
-
-  const style = Ok.of(
-    ElementDistinguishable.of(
-      [
-        ["border-width", "0px"],
-        ["font", "16px serif"],
-        ["color", "rgb(0% 0% 93.33333%)"],
-        ["outline", "0px"],
-        [
-          "background",
-          "linear-gradient(to right, rgb(97.64706% 97.64706% 94.5098%) 50%, rgb(0% 0% 100%) 50%)",
-        ],
-      ],
-      contrastPairings
-    )
-  );
-
-  t.deepEqual(await evaluate(ER62, { document }), [
-    passed(ER62, target, {
-      1: Outcomes.IsDistinguishable([style], [style], [style]),
-    }),
-  ]);
+  t.deepEqual(await evaluate(ER62, { document }), [inapplicable(ER62)]);
 });
 
-test(`evaluate() fails an <a> element that is not distinguishable from the <p> parent element as none of the foregrounds have a contrast of 3:1 in the parent element`, async (t) => {
+test(`evaluate() is inapplicable to an <a> element with a <p> parent element
+    when the parent element has the text content wrapped in a <span> element and it has the same foreground color as the <a> element`, async (t) => {
   const target = <a href="#">Link</a>;
 
   const document = h.document(
-    [<p>Hello {target}</p>],
+    [
+      <p>
+        <span>Hello</span>
+        {target}
```

Comment:
In this case since the `<p>`'s text content is wrapped into a `<span>`  I would only like to formulate something like:
```Javascript
if ( parentOfLink.hasNoDirectTextDescendant() && hasDifferentForeground(link, parentOfLink)) {
   containers = containers.set(link, parentOfLink);
   return yield link;
}
```
(lines 154-156 from the rules file)
where this bit `parentOfLink.hasNoDirectTextDescendant()` I might be able to use a common predicate to calculate?